### ISSUE TITLE
Low bitrate and poor video quality on Android phone.

### ISSUE THREAD

**Issue by gatecrasher777** [meetecho/janus-gateway#148]
This is related to issue #88 and the effect is the same before that issue was resolved by sending bitrate configuration messages near the start of the transmission.

On other webrtc projects I can get very good image quality using an android phone (Galaxy S5) with Chrome Beta.  But with Janus videoroom plugin, the image quality is very poor and the bitrate very low around 70kbs. I can get much higher bitrates in the echo test on your site, however, the echoed image just freezes (perhaps some other issue at play there).  Sending REMB bitrate messages also does not seem to work as they do from pcs and laptops, and Chrome's  webrtc-internals show that the max sendrate is stuck on the 300k videoroom plugin "default" as always seems to be before sending REMBs to adjust it.

I still don't know where this 300k default comes from, but it does seem to restrict Chrome from freely adjusting the bitrate to the capability of the connection. One doesn't see this ceiling in other webrtc projects when looking at the internals (or in echo test). I'm sure that whatever it is in the Janus code that sets this "default" is the root of the problem. 

So in summary, Chrome in other projects and Firefox, in general, seem to be calibrating bandwidth usage fine without intervention.  But in the videoroom plugin something is not quite right when using Chrome.  Sending REMBs mitigates this on a pc/laptop, but seems to have no effect on an Android device, resulting in very poor video quality.

**Comment by lminiero** [comment]
The 300k is an autoimposed value that Chrome decides. When REMB wasn't working as expected in the videoroom plugin (something I fixed in either issue 88 or somewhere else, can't remember) there was exactly a fixed 300k static value in webrtc-internals. I guess that in your mobile scenario the initial  REMB ramp-up sent by Janus is either ignored or not received. Have you tried sending a new configure with a different bitrate value during the call? Also check the value of NACK received, as in recent revisions Janus also sends NACKs whenever a packet is lost (which did indeed fix the freezes in the scenarios we experienced it).

**Comment by gatecrasher777** [comment]
Yes, my interface sends a bitrate configuration message using an algorithm based on screen resolution in the onplaying event. It repeats this several times in the first minute and then every minute thereafter. At 480x640 resolution it sends a max bitrate of about 570kbs, but Chrome's initial value doesn't change.

I'm not experiencing freezing in the videoroom, just poor video quality with this mobile. However, I will test this some more, perhaps with different media constraints to see if/how Chrome responds and check whether my initial observations are consistent.

**Comment by gatecrasher777** [comment]
Apologies for not getting back to this yet. Had a very busy week. Hopefully will get time to experiment this weekend.

**Comment by gatecrasher777** [comment]
Tested again on the latest build and quality was much, much better. I'm thinking this may also be due to improved nack management. Can close this issue.

**Comment by lminiero** [comment]
Good to know, thanks!
